### ISSUE TITLE
Test code for multiget; date_search to optionally do a multiget

### ISSUE THREAD

**Issue by tobixen** [python-caldav/caldav#115]
Pull request #111 needs some more test code and the date_search should have an optional parameter causing it to fetch the object data if needed, and by multiget if supported.

**Comment by tobixen** [comment]
https://tools.ietf.org/html/rfc4791#section-7.9

**Comment by tobixen** [comment]
Oups.  This functionality seems broken in master now.  It is important with test code!

**Comment by tobixen** [comment]
This also needs to be integrated with the sync work.  By now we have the following code there:

```python
        if load_objects:
            for obj in objects:
                try:
                    obj.load()
                except error.NotFoundError:
                    ## The object was deleted
                    pass
```

**Comment by tobixen** [comment]
Pushing this to 0.9.  0.8 is way overdue.

**Comment by tobixen** [comment]
This won't go into v0.10 either, I'm afraid.

**Comment by tobixen** [comment]
I think this was already partially done while dealing with issue #459

**Comment by tobixen** [comment]
I will add a functional test for multiget now.

multiget is already used instead of regular get when get fails.

mutliget could also be used from the `search`-method for calendar servers that does not follow the standards.  This would speed up collection of events a lot I suppose.  Will fork that out as a separate issue and deal with it after #402.
